backend/src/controllers/game_controller.ts

```diff
@@ -44,17 +47,47 @@ export class GameController extends Controller {
 
   @post()
   public async addGame(req: Request, res: Response): Promise<void> {
+    // needed for spotify
+    const token: string | undefined = req.header("Authorization");
+    if (token === undefined) {
+      throw new Error("Bearer authorization missing !");
+    }
+    const idSpotifyPlaylist: string | null = req.body.idSpotifyPlaylist || null;
+    if (idSpotifyPlaylist === null) {
+      throw new Error("Spotify id missing !");
+    }
+
+    let tracks: Array<Track>;
+    try {
+      tracks = await requestSpotifyTracks(token, idSpotifyPlaylist);
+    } catch (e) {
+      throw new Error("Error getting spotify tracks");
+    }
+
+    // save new game
+    const now: number = new Date().getTime();
     const game: Game = new Game();
-    game.startDate = req.body.startDate !== undefined ? toDate(req.body.startDate) : null;
+    game.startDate = req.body.startDate !== undefined ? toDate(req.body.startDate) || new Date(now + 30 * 1000) : new Date(now + 30 * 1000);
     game.isEnded = false;
     game.score = 0;
-    game.title = req.body.title || null;
-    game.questionsNumber = 0;
+    game.title = req.body.title || `game-${crypto.randomBytes(4).toString("hex")}`;
+    game.questionsNumber = tracks.length;
     game.isPublic = req.body.isPublic || false;
     game.mode = parseInt(req.body.mode, 10) || 0;
     game.idSpotifyPlaylist = req.body.idSpotifyPlaylist || null;
     await getRepository(Game).save(game);
-    res.sendJSON(game);
+
+    // todo create websocket room and store it in the game object.
+
+    // set new game session
+    game.tracks = tracks;
+    game.currentTrackIndex = -1;
+    req.gameSessions.new(game);
+
+    // send game without tracks
+    const userGame = JSON.parse(JSON.stringify(game));
```

Comment:
Why would you need that? ^^
If you want a copy of the object, you can do that instead:
```suggestion
    const userGame = {...game};
```

Reply:
Done